File: flutter_milibris_android/android/src/main/kotlin/com/milibris/flutter/FlutterMilibrisPlugin.kt
package com.milibris.flutter

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * Android Flutter plugin for flutter_milibris.
 *
 * This plugin exists to ensure the Milibris SDK dependencies are included
 * in the app's APK. All actual functionality is implemented in Dart via JNI.
 */
class FlutterMilibrisPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: flutter_milibris_ios/ios/Sources/flutter_milibris_ios/FlutterMilibrisPlugin.swift
import Flutter

/// Flutter plugin for flutter_milibris_ios.
///
/// This class exists only for plugin registration. All functionality is
/// implemented in Dart via FFI (@_cdecl functions in MilibrisReaderBridge.swift).
public class FlutterMilibrisPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {}
}
